### ISSUE TITLE
Limit chaincode container resource

### ISSUE THREAD

**Issue by yeasy** [hyperledger-archives/fabric#1846]
## Description

Currently we cannot limit the resource used by chaincode container. Hence the chaincode application can exhaust the system resource quite easily.

We may add this feature in the core.yaml, with supported in the fabric code. 
## Describe How to Reproduce

This is a feature discussion. If approved, I can help implement it.

**Comment by muralisrini** [comment]
@yeasy currently we have 1-1 relationship between number of chaincodes and containers. So the number of containers is somewhat constrained. 

But in general you are quite correct. Other future uses for containers (say to implement instances of a chaincode from the chaincode image) would could need us to consider bounds for containers. Or even currently there may be cases where the number of chaincodes themselves grow without bounds. It is not a bad thing to start thinking about how we'd put bounds on containers. 

Can you go into the details of how you would add support for this please ?

**Comment by mastersingh24** [comment]
@muralisrini  - Docker does not provide the greatest capabilities for constraining resources, but you can do it:   https://docs.docker.com/engine/reference/run/#runtime-constraints-on-resources

Looks like this is support in the go-dockerclient via [Config](https://godoc.org/github.com/fsouza/go-dockerclient#Config)

**Comment by muralisrini** [comment]
Thanks, @mastersingh24... I misread.  _Resources_  used by containers. Not _number_ of containers.

The difficulty would be coming up with reasonable defaults applicable to all chaincodes in an installation. Clearly one size won't fit all chaincode containers. 

However, we can place upperbounds ....  or maybe have users define resource classes (say classA/classB/classC/..) with different resource limits for each. Users can then associated a resource class for a chaincode.

**Comment by yeasy** [comment]
yes, i mean limiting the upper bound for chaincode container's resource.
We can never put it to production environment, unless we have that limit.
Before working on it, would like to listen to more options on this.
Thanks!
